src/LaunchDarkly.EventSource/MessageEvent.cs

```diff
@@ -46,87 +106,110 @@ public MessageEvent(string data, Uri origin) : this(data, null, origin)
 
         #region Public Properties
         /// <summary>
-        /// Gets the data received in the server sent event.
+        /// Gets the data received in the event as a string.
         /// </summary>
+        /// <remarks>
+        /// If the data was originally provided as a string, the same string is returned.
+        /// If it was provided as a UTF-8 byte array, the bytes are copied to a new string.
+        /// </remarks>
         /// <value>
         /// The data.
         /// </value>
-        public string Data
-        {
-            get
-            {
-                return _data;
-            }
-        }
+        public string Data => _dataString is null ? _dataUtf8Bytes.GetString() : _dataString;
+
+        /// <summary>
+        /// Gets the data received in the event as a UTF-8 byte span.
+        /// </summary>
+        /// <remarks>
+        /// If the data was originally provided as UTF-8 bytes, the returned value refers to
+        /// the same array, offset, and length (it is the caller's responsibility not to
+        /// modify the byte array). If it was originally provided as a string, the string
+        /// is copied to a new byte array.
+        /// </remarks>
+        public Utf8ByteSpan DataUtf8Bytes => _dataString is null ? _dataUtf8Bytes :
+            new Utf8ByteSpan(_dataString);
 
         /// <summary>
-        /// Gets the last event identifier received in the server sent event. This may be null if not provided by the API.
+        /// Gets the last event identifier received in the server-sent event. This may be null if not provided by the server.
         /// </summary>
         /// <value>
         /// The last event identifier.
         /// </value>
-        public string LastEventId
-        {
-            get
-            {
-                return _lastEventId;
-            }
-        }
+        public string LastEventId => _lastEventId;
 
         /// <summary>
-        /// Gets the origin URI of the server sent event.
+        /// Gets the origin URI of the stream that generated the server-sent event.
         /// </summary>
         /// <value>
         /// The origin.
         /// </value>
-        public Uri Origin
-        {
-            get
-            {
-                return _origin;
-            }
-        }
+        public Uri Origin => _origin;
+
+        /// <summary>
+        /// True if the event data is stored internally as UTF-8 bytes.
+        /// </summary>
+        /// <remarks>
+        /// The data can be accessed with either <see cref="Data"/> or <see cref="DataUtf8Bytes"/>
+        /// regardless of the value of this property. The property only indicates the <i>original</i>
+        /// format of the data, so, for instance, if it is <see langword="true"/> then
+        /// reading <see cref="Data"/> will have more overhead (due to copying) than
+        /// reading <see cref="DataUtf8Bytes"/>.
+        /// </remarks>
```

Comment:
Same as above for not having a `<value>` tag. That said, I've not looked at the generated code doc for this API, I'm not sure how important it is to have that tag.

Reply:
I've never been convinced that there is any point to using the `<value>` tag. You end up with a bunch of doc comments that look like this:

```
/// <summary>The number of items.</summary>
/// <value>the number of items</value>
int Count { get; }
```

I mean, it's a property, it doesn't have any parameters, so I can't think of any circumstance in which the `<value>` text wouldn't just be exactly the same as the `<summary>`— or a slightly shorter version of, but why would that matter when the `<summary>` is what you see _first_ in the docs?

Reply:
Yeah, I've been basically thinking the same, I just hadn't looked at the generated docs to see if it seemed important to how they turn out. Do you know if it provides any value in IDEs that `summary` does not cover?

Reply:
No, I've never found any reason at all that it's helpful. IDEs that show you a doc comment as hover text, etc., will show the `<summary>`.

Reply:
Alright then, I'll keep that in mind when reading dotnet doc comments then. They just stood out as an inconsistency to me.

Reply:
I mean, to me, even the official documentation for this tag doesn't make a case for it:

https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/value

Like, if the only reason to have a separate tag is to describe _an implementation detail that the caller shouldn't even know about_ (in this example, they're explaining the private backing field for the property) then why would this be a part of generated API documentation?

Reply:
Also worth noting: when doc generation is turned on, the compiler spits out warnings if you omit a `<summary>` or `<param>` or `<returns>`, but it doesn't bother if you omit `<value>`.